File: DP/boj_11057_.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		int num = Integer.parseInt(br.readLine());
		long[][] arr = new long[num+1][10];
	
		for (int i = 0; i <= 9; i++) {
			arr[0][i] = 1;
		}
		
		for (int i = 1; i <= num; i++) {
			for (int j = 0; j <= 9; j++) {
				for (int k = j; k <= 9; k++) {
					arr[i][j] += arr[i-1][k];
					arr[i][j] %= 10007;
				}
			}
		}
		
		System.out.println(arr[num][0] % 10007);
	}	
}

File: DP/boj_11057.py
num = int(input())

arr = [[0] * 10 for _ in range(num + 1)]
mod = 10007
int(mod)

for i in range(0, 10):
    arr[0][i] = 1

for i in range(1, num+1):
    for j in range(0, 10):
        for k in range(j, 10):
            arr[i][j] += arr[i-1][k]
            arr[i][j] %= mod
    
print(arr[num][0] % mod)

